azure-iot-device/tests/common/pipeline/test_pipeline_stages_base.py

```diff
@@ -252,6 +252,14 @@ def stage(self, mocker, cls_type, init_kwargs):
 )
 
 
+only_disconnected = [pytest.param(False, id="Pipeline Disconnected")]
+only_connected = [pytest.param(True, id="Pipeline Connected")]
+connected_and_disconnected = [
+    pytest.param(True, id="Pipeline Connected"),
+    pytest.param(False, id="Pipeline Disconnected"),
+]
+
+
 @pytest.mark.describe(
     "AutoConnectStage - .run_op() -- Called with an Operation that requires an active connection"
 )
```

Comment:
Now that we've changed to create two unique flows in this circumstance, (one where we think we're connected, and one where we think we're not) it might be a good idea to split this up into two different testing units for clarity.

Reply:
Good idea.  that might help with my it() phrasing problem.

---
In reply to: [388044172](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r388044172) [](ancestors = 388044172)

Reply:
Yeah, that's usually the giveaway that something needs to change.

---
In reply to: [388477549](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r388477549) [](ancestors = 388477549,388044172)